ethereum/networks/src/main/java/tech/pegasys/teku/networks/StateBoostrapConfig.java

```diff
@@ -13,32 +13,72 @@
 
 package tech.pegasys.teku.networks;
 
+import static com.google.common.base.Preconditions.checkNotNull;
+
+import java.util.Optional;
+import tech.pegasys.teku.infrastructure.http.UrlSanitizer;
+
 public class StateBoostrapConfig {
+  public static final String FINALIZED_STATE_URL_PATH = "eth/v2/debug/beacon/states/finalized";
+  public static final String GENESIS_STATE_URL_PATH = "eth/v2/debug/beacon/states/genesis";
+
+  private Optional<String> genesisState = Optional.empty();
+  private Optional<String> initialState = Optional.empty();
+  private Optional<String> checkpointSyncUrl = Optional.empty();
   private boolean isUsingCustomInitialState;
-  private boolean isUsingCheckpointSync;
   private boolean allowSyncOutsideWeakSubjectivityPeriod;
 
   StateBoostrapConfig() {}
 
-  public void setUsingCustomInitialState(final boolean usingCustomInitialState) {
-    isUsingCustomInitialState = usingCustomInitialState;
+  public void setGenesisState(final String genesisState) {
+    checkNotNull(genesisState);
+    this.genesisState = Optional.of(genesisState);
+  }
+
+  public void setInitialState(final String initialState) {
+    checkNotNull(initialState);
+    this.initialState = Optional.of(initialState);
+    this.isUsingCustomInitialState = false;
+  }
+
+  public void setCustomInitialState(final String initialState) {
+    checkNotNull(initialState);
+    this.initialState = Optional.of(initialState);
+    this.isUsingCustomInitialState = true;
```

Comment:
it looks a bit fragile  that it could be damaged by the call order. Maybe set default to false and override it only here and don't override in `setInitialState`?